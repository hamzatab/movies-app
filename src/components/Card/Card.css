.card {
    width: 325px;
    margin: 1rem;
    position: relative;
    transition: all ease-in-out 0.3s;
    cursor: pointer;
}

.cardImage {
    height: 430px;
    background-color: beige;
    overflow-y: hidden;
    position: relative;
    background-color: black;
}

.cardImage>img {
    width: 100%;
    transition: all ease-in 0.2s;
}


.cardOverview {
    top: 0;
    opacity: 0;
    height: 100%;
    padding: 0.5rem;
    position: absolute;
    color: white;
    background-color: #333533;
    transition: all ease-in 0.3s;
}

.card:hover .cardImage>img {
    opacity: 0.3;
    transform: scale(1.1);
}

.card:hover .cardOverview {
    opacity: 1;
}

.cardHeader {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 1rem;
    background-color: black;
    color: white;
}

.cardHeaderName {
    font-weight: bold;
    font-size: 1rem;
}

.cardHeaderRating {
    font-weight: bold;
    padding: 5px 10px;
    background-color: #343533;
    color: red;
    border-radius: 10px;
}