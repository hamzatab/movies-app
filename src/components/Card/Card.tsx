
import Movie from "../../interfaces/Movie";
import "./Card.css"


export const Card = (movie: Movie) => {

  return (
    <div className="card" key={movie.id}>
       <div className="cardHeader">
        <div className="cardHeaderName">
          {movie.name} ({new Date(movie.releaseDate).getFullYear()})
        </div>
        <div className="cardHeaderRating">{movie.rate}</div>
      </div>
      <div className="cardImage">
        <img src={movie.poster} alt={movie.name} />
        <div className="cardOverview">{movie.overview}</div>
      </div>
     
    </div>
  );
};
