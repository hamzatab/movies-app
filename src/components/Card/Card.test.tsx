import "@testing-library/jest-dom"
import { render } from "@testing-library/react"
import { Card } from "./Card"

const params = {
    id: 1,
    releaseDate: "2022",
    name: "test movie",
    rate: 8,
    overview: "overview",
    poster: undefined,
    language: "fr"
}


test('renders correctly', () => {
    const { asFragment } = render(<Card  {...params} />)
    expect(asFragment()).toMatchSnapshot();
})

