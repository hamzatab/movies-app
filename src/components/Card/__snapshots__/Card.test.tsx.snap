// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="cardHeader"
    >
      <div
        class="cardHeaderName"
      >
        test movie (2022)
      </div>
      <div
        class="cardHeaderRating"
      >
        8
      </div>
    </div>
    <div
      class="cardImage"
    >
      <img
        alt="test movie"
      />
      <div
        class="cardOverview"
      >
        overview
      </div>
    </div>
  </div>
</DocumentFragment>
`;
